SWIPs/swip-20-honey_to_money.md

```diff
@@ -0,0 +1,103 @@
+---
+SWIP: 20
+Title: Honey to Money Oracle
+Author: Aron Fischer <aron@ethswarm.org>, Deigo Masini <dmasini@iovlabs.org>, Fabio Barone <fabio@ethswarm.org>, Marcelo Ortelli <mortelli@infuy.com>, Rinke Hendriksen <rinke@ethswarm.org>, Vojtěch Šimetka <vojtech@iovlabs.org>
+Discussions-to: <https://swarmresear.ch/>
+Status: Draft
+Type: Standards track
+Category: Core
+Created: 30-08-2019
+---
+<!--You can leave these HTML comments in your merged SWIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new SWIPs. Note that a SWIP number will be assigned by an editor. When opening a pull request to submit your SWIP, please use an abbreviated title in the filename, `SWIP-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+ 
+## Simple summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SWIP.-->
+Swarm needs a mechanism to set prices which makes sense relative to other (distributed storage) services in the market. As the absolute price of using Swarm is defined by the cost of honey (Swarm's internal accounting unit), this SWIP proposes a minimal interface to enable upgrading honey prices, network-wide and for all nodes. The specifics on what an efficient price should reflect and how the update process looks like is *not* part of this SWIP.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+It is expected that the price of using Swarm will change over time to reflect the changing economic landscape as well as to facilitate experimentation. Such price updates must happen atomically to prevent accounting unbalances from appearing. To lay the groundwork for enabling such price changes, this SWIP proposes a minimal interface for a *price-oracle*. Furthermore, this SWIP includes a proposal for implementing an *on-chain* price oracle, governed by Swarm developers and stakeholders to allow for experimentation.
+ 
+This SWIP is part of a series of SWIPs (but can be implemented on its own). To see the full picture, please refer to [swip-message_to_honey](https://github.com/Eknir/SWIPs/blob/SWIP-message_to_honey/SWIPs/swip-message_to_honey.md), [swip-multiple_payment_processing_support](https://github.com/Eknir/SWIPs/blob/SWIP-multiple_payment_processing_support/SWIPs/swip-multiple_payment_processing_support.md) and the diagram below:
+![SWIP_Diagrams.png](./../assets/swip-honey_to_money/SWIP_Diagrams.png)
```

Comment:
To the diagram:

* Do lines terminating outside nodes have any significant meaning as opposed to lines intersecting the nodes?

* Why is there a line connecting process message, disconnect threshold reached condition (no), update balance. The "no" seems to go in both directions, and terminate nowhere.

Reply:
> Do lines terminating outside nodes have any significant meaning as opposed to lines intersecting the nodes?

What do you mean here?

> Why is there a line connecting process message, disconnect threshold reached condition (no), update balance. The "no" seems to go in both directions, and terminate nowhere.

Sorry for the confusion here. When I checked it just now, it also took me a while to figure out, meaning it was not clear. Process message is an end-state and can only be reached via `should it be accounted for, no`, `update balance` (debitor) and `disconnect threshold reached, no` (creditor).

`Disconnect threshold reached, no => process message` is the only allowed direction. I will have a look together with @vojtechsimetka how we can make this clear. Thanks.

> Please consider three separate diagrams under each other; it's too small to be legible in the markdown document, and the enlargement behind the first click is a frame with scroll.

We will have a look at how this can be improved.

Reply:
> Why is there a line connecting process message, disconnect threshold reached condition (no), update balance. The "no" seems to go in both directions, and terminate nowhere.

Nice catch! Should be good now